### PR TITLE
[Core] Faster logit_bias_logits_processor

### PR DESCRIPTION
**Pull request by xu-song**
## Changes

- Change python ops to tensor ops, which reduce time cost of `logit_bias_logits_processor` from `106ms` to `0.01ms` (when `len(logit_bias)=10000`).

| Before(python ops) | After (tensor ops) |
|-----------------|--------|
| <pre>for token_id, bias in logit_bias.items():<br/>  logits[token_id] += bias </pre>|`logits.index_add_(0,logit_bias_index, logit_bias_value)`|

- Add cache for logit_bias

## Impl history
- [**v1**](https://github.com/vllm-project/vllm/blob/a50b0ead8084692522ce41c65ee7ada4002ed29d/vllm/entrypoints/openai/logits_processors.py#L50-L51): no cache
- [**v2**](https://github.com/vllm-project/vllm/blob/62a74e3d1bf3b9b34d17cd5fb46e36f0be756d89/vllm/entrypoints/openai/logits_processors.py#L50-L53): cache in single request
- [**v3**](https://github.com/vllm-project/vllm/blob/cd9f33fa750fbeabfd8ee8526b5a52ae99580f28/vllm/entrypoints/openai/logits_processors.py#L70-L74): cache across different requests

## Performance

| len(logit_bias) | time cost (ms) <br> `before -> v1 -> v2/v3`  |
|-----------------|--------|
| 1              |  4.5 -> 0.3 -> 0.01 |
| 20              |  4.5 -> 0.3 -> 0.01 |
| 100             |   5.3 -> 0.3 -> 0.01   |
| 1000             |    14.4 -> 0.3 -> 0.01     |
| 10000             |    106 -> 0.4 -> 0.01    |

experiment settings: 
```
GPU: A100 
model: Llama-3.2-1B-Instruct
```

### CODE REVIEW

**Comment by imkero** [comment]
If `len(logit_bias)` is large, maybe we can keep a copy of `logit_bias["index"]` and `logit_bias["value"]` in the **device-side memory** ahead of time, to avoid tensor copy across each sample step?

**Comment by xu-song** [comment]
> If `len(logit_bias)` is large, maybe we can keep the copy of `logit_bias["index"]` and `logit_bias["value"]` in the device memory ahead of time (or in the first sample step), and re-use it in the following sample steps, to avoid duplicated tensor copy?

Yes, a new commit has been added, which avoid duplicated tensor copy

**Comment by xu-song** [comment]
@zhuohan123 @youkaichao Is there anyone who can review this pr?

**Comment by xu-song** [comment]
@njhill conflicts have been resolved.
